theos 2.6 (new formula)

Ishaanahuja7 opened the pull request:
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

SMillerDev commented on Formula/theos.rb:
```diff
@@ -0,0 +1,45 @@
+  end
+
+  def caveats
+    <<~EOS
+      To use Theos, please put the following in your .profile:
+        export THEOS="#{opt_prefix}"
+    EOS
```
We should probably use `wrapper_script` to do this for people

Ishaanahuja7 replied:
So just use echo to add this line to .profile/.zshrc/.bashrc?

SMillerDev replied:
No, use `wrapper_script`

Ishaanahuja7 replied:
Cant find any documentation on what wrapper_script is and how to use it. Do you have a link?

SMillerDev replied:
https://rubydoc.brew.sh/Pathname.html#write_env_script-instance_method for example

Ishaanahuja7 replied:
Thanks! So theos is not just 1 command/file, there are multiple files which require the THEOS env variable and it would just be a lot easier for user to set that in their .profile rather than us making a wrapper script of almost all executables. What do you think?

SMillerDev replied:
There is https://rubydoc.brew.sh/Pathname.html#env_script_all_files-instance_method for wrapping all files. People rarely read the caveats so anything that needs can be done automatically should be done automatically

Ishaanahuja7 replied:
It kinda wont work I'm realizing that some files in bin dir are symlinks to files outside of bin dir.

SMillerDev commented on Formula/theos.rb:
```diff
@@ -0,0 +1,45 @@
+    version "2.6"
+  end
+
+  keg_only "symlinking Theos makes it unusable"
```
This should be reported and fixed upstream

Ishaanahuja7 replied:
Packaging theos is not a priority for the developers it looks like so will have to keep it keg_only for now. Users can add the bin to path, so its still usable

SMillerDev replied:
Well, Homebrew/core doesn't exist to package everything. If the developers don't want their software packaged we can comply with that.

Ishaanahuja7 replied:
They haven't said it explicitly, I'm guessing based on the issue responses in their repo. But checks are passing now, changed keg_only message

SMillerDev replied:
You'll have to get them to fix it because this makes the formula a lot less user friendly. With this issue I would not consider the software fit for homebrew-core.

Ishaanahuja7 replied:
Okay, let's keep the pull request open until they fix it then?

SMillerDev commented on Formula/theos.rb:
```diff
@@ -0,0 +1,45 @@
+  skip_clean "lib", "sdks"
+
+  resource "sdks" do
+    url "https://github.com/theos/sdks.git",
+      branch:   "master",
+      revision: "fd931ca723ee46994ad9f73c2f0929d1ab9732ca"
+  end
```
I don't think we can ship software under XCode's license

Ishaanahuja7 replied:
Removed sdks resource and added in caveats

carlocab replied:
Users having to install this manually probably goes against https://docs.brew.sh/Acceptable-Formulae#stuff-that-requires-heavy-manual-prepost-install-intervention.

carlocab reviewed (changes requested):
I don't think this is a good fit for a Homebrew formula.

1. This software [updates itself](https://github.com/theos/theos/blob/e53ec51e92a3716def537c7ff1b5b2a86930700c/bin/update-theos). https://docs.brew.sh/Acceptable-Formulae#we-dont-like-tools-that-upgrade-themselves
2. It requires installing files into `prefix/"sdks"`. These files will get deleted by `brew upgrade theos`.

Ishaanahuja7 commented:
@carlocab I removed the update scripts, and fixed the sdks issues by placing them in its etc folder which prefix/"sdks" symlinks to, so when homebrew upgrades the package, etc/sdks wont get deleted and new version will still point to etc/sdks

Ishaanahuja7 commented:
Still checking if keg_only can be removed with couple patches

Ishaanahuja7 commented:
@carlocab Removed keg_only, found simple patch

carlocab commented on Formula/theos.rb:
```diff
@@ -0,0 +1,46 @@
+  license "GPL-3.0-or-later"
+  head "https://github.com/theos/theos.git", branch: "master"
+
+  stable do
+    url "https://github.com/theos/theos.git",
+      branch:   "master",
+      revision: "e53ec51e92a3716def537c7ff1b5b2a86930700c"
+    version "2.6"
+  end
```
You don't need a `stable` block for this. Also, we only use tagged releases. It looks like this is just an arbitrary commit. See https://docs.brew.sh/Acceptable-Formulae#stable-versions.

carlocab reviewed (commented):
While you've fixed some of my concerns, I'm still of the opinion that this isn't a good fit for a formula. If you really want to have one, it's probably better that you host it in an external tap.

Here are some docs to help you get started:

https://docs.brew.sh/Taps
https://docs.brew.sh/Interesting-Taps-and-Forks
https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/

SMillerDev commented:
With 2 maintainers expressing their concerns about the fit with Homebrew/core I think we should just close this.
